src/js/jsx/sections/style/Radius.jsx

```diff
@@ -40,72 +40,130 @@ define(function (require, exports, module) {
     var Radius = React.createClass({
         mixins: [FluxMixin, Coalesce],
 
+        /**
+         * Keeps track of the radius on Scrub start
+         *
+         * @type {number}
+         */
+        _initialScrubRadius: null,
+
         propTypes: {
             document: React.PropTypes.object.isRequired
         },
 
-        shouldComponentUpdate: function (nextProps) {
+        shouldComponentUpdate: function (nextProps, nextState) {
+            return this.state !== nextState;
+        },
+
+        componentWillReceiveProps: function (nextProps) {
             var getRelevantProps = function (props) {
-                var layers = props.document.layers.selected;
+                    var layers = props.document.layers.selected.filter(function (layer) {
```

Comment:
It is unfortunate that we filter "selected layers with radii" here, and then may immediately do it again in `_calculateState`.  Adding this subset of layers to layerstructure might help, but may be overkill.  Mentioning it anyway, for kicks.